src/mbgl/style/layers/symbol_layer.cpp

```diff
@@ -125,7 +125,7 @@ void SymbolLayer::setIconOptional(PropertyValue<bool> value) {
     impl->layout.iconOptional.set(value);
 }
 PropertyValue<AlignmentType> SymbolLayer::getDefaultIconRotationAlignment() {
-    return { AlignmentType::Viewport };
+    return { AlignmentType::Undefined };
```

Comment:
Note: this file is generated so we will need to update [generate-style-code.js](https://github.com/mapbox/mapbox-gl-native/blob/master/scripts/generate-style-code.js) to make the required change here (which will probably be to make this [conform to the new `auto` value](https://github.com/mapbox/mapbox-gl-style-spec/pull/493)